docs/architecture/modernize-desktop/example-migration-to-net-core.md

```diff
@@ -0,0 +1,365 @@
+using the designer, you must create a new .NET Core project in parallel with the
+old one and share assets. When it is time to modify UI elements in the designer
+you can switch to the old project to do that, and since assets are linked there
+will be updated in the .NET Core project.
+
+.NET Core SDK style project is a lot simpler that .NET Framework and apart from
+the mentioned PackageReference stuff, you will not need to do much more. SDK
+style adds all necessary files underneath project file location like `.cs` and
+`.xaml` files without the need to explicitly include them in the `.csproj`.
+
+#### Assembly.info considerations
+
+Attributes are auto generated on .NET Core projects. If the project contains
+`AssemblyInfo.cs` file, the definitions will be duplicated, which will cause a
+compilation conflicts. You can delete the older `AssemblyInfo.cs` file or
+disable auto-generation by adding this entry to the .NET Core project file:
+
+```xml
+<GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+```
+
+#### Resources
+
+Embedded resources are included automatically but resources are not, so you need
+to migrate those to the new .csproj.
+
+#### Packages References
+
+With the "Migrate packages.config to PackageReference" option, you can easily
+move your external packages references to the new format as mentioned above.
+
+#### Update package references
+
+Proceed to update the version of the packages you have found to be compatible as
+depicted in previous section.
+
+### Fix Code and Build
+
+#### Microsoft.Windows.Compatibility
+
+In case your applications depends on APIs that aren't available on .NET Core like
+Registry, ACLs, WCF you have to include a reference to the
+Microsoft.Windows.Compatibility package to add these Windows-specific APIs. They
+work on .NET Core but aren't included as they aren't cross-platform.
+
+There is a tool called API Analyzer
+(<https://docs.microsoft.com/en-us/dotnet/standard/analyzers/api-analyzer>) that
+help you identify not compatible APIs as long as you develop your code.
+
+#### Use \#if defines
+
+If you need different execution paths when targeting .NET Framework and .NET
+Core, you should use compilation constants and code some \#if statements to keep
+the same code base for both targets.
+
+#### Technologies not available on .NET Core
+
+Some technologies aren't available on .NET Core like:
+
+* AppDomains
+* Remoting
+* Code Access Security
+* WCF Server
+* Windows Workflow
+
+Therefore, it is time to look for a replacement for those.
+
+#### Regenerate autogenerated clients
+
+If your application uses some autogenerated code like, for example, a WCF
+client, you may need to regenerate this code to target .NET Core. Sometimes you
+can find some missing references since it may not be included as default as part
+of the basic .Net Core assemblies. Using a tool like <https://apisof.net/> you
+can easily locate the assembly the missing reference lives in and add it from
+NuGet.
```

Comment:
```suggestion
If your application uses autogenerated code, such as a WCF
client, you may need to regenerate this code to target .NET Core. Sometimes, you
can find some missing references since it may not be included as default as part
of the basic .NET Core assemblies. Using a tool like <https://apisof.net/>, you
can easily locate the assembly the missing reference lives in and add it from
NuGet.
```